### PR TITLE
feat(plugins): Add deprecation warning in UI

### PR DESCRIPTION
**Pull request by ceorourke**
Have mercy on my backend soul. This is the front end piece to https://github.com/getsentry/sentry/pull/28137 which is a temporary banner to display plugin deprecation dates. The "upgrade now" button takes the user to the first party or sentry app version of the plugin so they can install that instead.

Screen Shot 2021-08-24 at 11 35 12 AM

Screen Shot 2021-08-23 at 5 37 46 PM

### CODE REVIEW

**Comment by robinrendle** [comment]
@ceorourke can we left align the alert in the first screenshot 😬

File: static/app/types/index.tsx [hunk 1]
```diff
@@ -612,6 +612,9 @@ export type PluginProjectItem = {
 
 export type PluginWithProjectList = PluginNoProject & {
   projectList: PluginProjectItem[];
+  altIsSentryApp: boolean;
```
Comment (mgaeta) [inline static/app/types/index.tsx:615]: Each of these are optional so I'd make them 
```javascript
altIsSentryApp?: boolean;
deprecationDate?: string;
firstPartyAlternative?: string;
```

File: static/app/views/organizationIntegrations/integrationListDirectory.tsx [hunk 2]
```diff
@@ -380,7 +380,17 @@ export class IntegrationListDirectory extends AsyncComponent<
...
   renderPlugin = (plugin: PluginWithProjectList) => {
     const {organization} = this.props;
-
+    let upgradeUrl = '';
+    let deprecationText = '';
+    if (plugin.deprecationDate !== null) {
+      deprecationText = `This integration is being deprecated on ${plugin.deprecationDate}. Please upgrade to avoid any disruption.`;
+    }
+    if (plugin.firstPartyAlternative !== null) {
+      upgradeUrl =
```
Comment (mgaeta) [inline static/app/views/organizationIntegrations/integrationListDirectory.tsx:389]: I don't know if this is actually clearer but it's less code.
```javascript
const resource = plugin.altIsSentryApp ? 'sentry-apps' : 'integrations';
upgradeUrl = `/settings/${organization.slug}/${resource}/${plugin.firstPartyAlternative}/`;
```

**Review by mgaeta — CHANGES_REQUESTED** [review]
I think you should try to extract this component to a file rather than repeat it twice.
```javascript
import React, {Component} from 'react';
import styled from '@emotion/styled';

import Alert from 'app/components/alert';
import Button from 'app/components/button';
import {IconWarning} from 'app/icons';
import {t} from 'app/locale';
import space from 'app/styles/space';
import {Organization, PluginWithProjectList} from 'app/types';
import {trackIntegrationEvent} from 'app/utils/integrationUtil';

type Props = {
  organization: Organization;
  plugin: PluginWithProjectList;
};

type State = {};

class PluginDeprecationAlert extends Component<Props, State> {
  render() {
    const {organization, plugin} = this.props;

    // Short-circuit if not deprecated.
    if (!plugin.deprecationDate) {
      return <React.Fragment />;
    }
    let upgradeUrl = '';
    if (plugin.firstPartyAlternative !== null) {
      upgradeUrl =
        plugin.altIsSentryApp === false
          ? `/settings/${organization.slug}/integrations/${plugin.firstPartyAlternative}/`
          : `/settings/${organization.slug}/sentry-apps/${plugin.firstPartyAlternative}/`;
    }

    return (
      <AlertContainer>
        <Alert type="warning" icon={<IconWarning size="sm" />}>
          <span>{`This integration is being deprecated on ${plugin.deprecationDate}. Please upgrade to avoid any disruption.`}</span>
          <ResolveNowButton
            href={`${upgradeUrl}?tab=configurations&referrer=directory_upgrade_now`}
            size="xsmall"
            onClick={() =>
              trackIntegrationEvent('integrations.resolve_now_clicked', {
                integration_type: 'plugin',
                integration: plugin.slug,
                organization,
              })
            }
          >
            {t('Upgrade Now')}
          </ResolveNowButton>
        </Alert>
      </AlertContainer>
    );
  }
}

const ResolveNowButton = styled(Button)`
  color: ${p => p.theme.subText};
  float: right;
`;

const AlertContainer = styled('div')`
  padding: 0px ${space(3)} 0px 68px;
`;

export default PluginDeprecationAlert;
```

**Review by mgaeta — APPROVED** [review]

**Pull request merged**